spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/factory/RewriteRequestParameterGatewayFilterFactory.java

```diff
@@ -59,14 +63,25 @@ public Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
 				ServerHttpRequest req = exchange.getRequest();
 
 				UriComponentsBuilder uriComponentsBuilder = UriComponentsBuilder.fromUri(req.getURI());
-				if (req.getQueryParams().containsKey(config.getName())) {
-					uriComponentsBuilder.replaceQueryParam(config.getName(), config.getReplacement());
+
+				MultiValueMap<String, String> queryParams = new LinkedMultiValueMap<>(req.getQueryParams());
+				if (queryParams.containsKey(config.getName())) {
+					queryParams.remove(config.getName());
+					queryParams.add(config.getName(), config.getReplacement());
 				}
```

Comment:
The `ServerHttpRequest.getQueryParams()` method returns the `decoded query parameters`. 
Therefore, it can replace `new parameter` in the `decoded query parameters`.